### ISSUE TITLE
[bug] Runnable target isn't distributed

### ISSUE THREAD

**Issue by rcrowe** [moonrepo/moon#1407]
**Describe the bug**

After the target `auth/jwks-publisher:docker-build` is listed in `Gathering runnable targets`, none of the jobs list it under `Distributing targets across jobs`.
`:docker-build` is an inherited task, that runs for other projects, but not for `auth/jwks-publisher`.

Checking `moon task auth/jwks-publisher` looks fine, the `moon.yml` for the project doesn't look to be significantly different to those projects working & `moon ci` suggests it's triggered.

I'm not sure if this is because I've accidentally (need to check) cached something under `.moon/cache` that I should not have?

**Steps to reproduce**

1) `Gathering runnable targets` lists `auth/jwks-publisher:docker-build`

<img width="553" alt="Screenshot 2024-03-29 at 22 35 02" src="https://github.com/moonrepo/moon/assets/282571/56c307a6-5f9f-4158-9409-f02e855e5400">

2) Under the 4 jobs, `Distributing targets across jobs` does not list `auth/jwks-publisher:docker-build`

<img width="578" alt="Screenshot 2024-03-29 at 22 44 20" src="https://github.com/moonrepo/moon/assets/282571/233bd420-6e34-4239-b3db-0d701e802bcd">
<img width="541" alt="Screenshot 2024-03-29 at 22 44 38" src="https://github.com/moonrepo/moon/assets/282571/ac61a434-d38e-46d8-a93a-565e23b06d3c">
<img width="529" alt="Screenshot 2024-03-29 at 22 44 58" src="https://github.com/moonrepo/moon/assets/282571/6fd52f6b-23a5-41ba-9317-7d429bf37fd1">
<img width="529" alt="Screenshot 2024-03-29 at 22 45 12" src="https://github.com/moonrepo/moon/assets/282571/6b8f6b45-a5c4-45f8-907c-a31d1f461954">

**Comment by rcrowe** [comment]
I've just tried `git commit --allow-empty -m "Trigger" && git push origin head`. The runs from that moon looked to work as I expected it. 

My assumption at this point was the prior run that didn't work updated a file in the cache directory, so this subsequent call worked.

I've made sure to follow [the docs](https://moonrepo.dev/docs/guides/ci#manual-persistence) and cache just `.moon/cache/{hashes,outputs}`. 

Am I correct that the issue I saw was because of some out of sync file? Not sure I've seen this locally before 🤔
